keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/KDEVertex.java

```diff
@@ -0,0 +1,135 @@
+package io.improbable.keanu.vertices.dbl.probabilistic;
+
+
+import io.improbable.keanu.distributions.continuous.Gaussian;
+import io.improbable.keanu.distributions.continuous.Uniform;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class KDEVertex extends ProbabilisticDouble {
+    public double bandwidth;
+    private DoubleTensor samples;
+
+    public KDEVertex(DoubleTensor samples, double bandwidth){
```

Comment:
Does this vertex have no parents?

Reply:
It doesn't. This vertex should be used to approximate, for example, a posterior distributions based on samples drawn from it.

Reply:
I see - so is it intended to be used as the root of a graph, i.e. it will have children?

Reply:
Yep, it may be used as an input to other vertices.